File: benchmarks/tsb/bench_dataframe_skew_kurt.ts
/**
 * Benchmark: skewDataFrame / kurtDataFrame — skewness and kurtosis on a 10k×10 DataFrame.
 * Outputs JSON: {"function": "dataframe_skew_kurt", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { DataFrame, skewDataFrame, kurtDataFrame } from "../../src/index.ts";

const ROWS = 10_000;
const COLS = 10;
const WARMUP = 5;
const ITERATIONS = 20;

const columns: Record<string, number[]> = {};
for (let c = 0; c < COLS; c++) {
  columns[`col${c}`] = Array.from({ length: ROWS }, (_, i) => Math.sin((i + c) * 0.01) * 100);
}
const df = new DataFrame(columns);

for (let i = 0; i < WARMUP; i++) {
  skewDataFrame(df);
  kurtDataFrame(df);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  skewDataFrame(df);
  kurtDataFrame(df);
}
const total = performance.now() - start;

console.log(JSON.stringify({ function: "dataframe_skew_kurt", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));

File: benchmarks/pandas/bench_dataframe_skew_kurt.py
"""
Benchmark: DataFrame.skew() / DataFrame.kurt() — skewness and kurtosis on a 10k×10 DataFrame.
Outputs JSON: {"function": "dataframe_skew_kurt", "mean_ms": ..., "iterations": ..., "total_ms": ...}
"""
import json
import math
import time
import pandas as pd
import numpy as np

ROWS = 10_000
COLS = 10
WARMUP = 5
ITERATIONS = 20

data = {f"col{c}": np.array([math.sin((i + c) * 0.01) * 100 for i in range(ROWS)]) for c in range(COLS)}
df = pd.DataFrame(data)

for _ in range(WARMUP):
    df.skew()
    df.kurt()

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.skew()
    df.kurt()
total = (time.perf_counter() - start) * 1000

print(json.dumps({"function": "dataframe_skew_kurt", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))
